quantumcat/circuit/circuit.py

```diff
@@ -357,34 +1112,96 @@ def mcxrec_gate(self, control_qubits, target_qubit):
         return self
 
     def mcp_gate(self, lam, control_qubits, target_qubit):
+        """
+        Uses multiple control qubits to decide the application of the P gate to the target qubit.
+        If all the control qubits are in the |1> state, the P gate is performed
+        on the target qubit. Otherwise, the the P gate is not performed on the target qubit.
+
+        Args:
+            lam: the angle(in radian) to be rotated.
+            control_qubits: list of control qubit indices, to which the gate is to be applied.
+            target_qubit: index of target qubit, to which the gate is to be applied.
+
+        Returns:
+            self: modified state of the qubit to the circuit.
+
+        Raises:
+            CircuitError: if qubit index is not within range.
+        """
         self.check_qubit_boundary(target_qubit)
         for qubit in control_qubits:
             self.check_qubit_boundary(qubit)
         self.operations.append(
             {OpType.mcp_gate: control_qubits[:] + [target_qubit], constants.PARAMS: [lam, len(control_qubits)]})
         return self
 
-    def mct_gate(self, control_qubits, target_qubit, ancilla_qubits=None, mode='noancilla'):
-        # self.check_qubit_boundary(control_qubit1)
-        self.check_qubit_boundary(target_qubit)
-        self.operations.append({OpType.mct_gate: [control_qubits, [target_qubit], ancilla_qubits, mode]})
+    def mct_gate(self, control_qubits, target_qubits, ancilla_qubits=None, mode='noancilla'):
+        """
+        Uses multiple control qubits to decide the application of the X gate to multiple target qubits.
+        If all the control qubits are in the |1> state, the X gate is performed
+        on the target qubits. Otherwise, the the X gate is not performed on the target qubits.
+
+        Args:
+            control_qubits: list of control qubit indices, to which the gate is to be applied.
+            target_qubits: list of target qubit indices, to which the gate is to be applied.
```

Comment:
ancilla_qubits isn't mentioned in Args:, also confirm if 'mode' parameters needs to mentioned in Args: